tools/resourcedocsgen/pkg/util/language/language.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package language exposes an enumeration of docs generation target languages:
+//
+// - [CSharp]
+// - [Go]
+// - [Java]
+// - [Python]
+// - [NodeJS]
+// - [YAML]
+//
+// For an iterator over all valid languages, use [All].
+package language
+
+import "iter"
+
+// Language is a valid docsgen language. The default value is not valid.
+type Language struct{ tag string }
+
+var (
+	CSharp = Language{"csharp"}
+	Go     = Language{"go"}
+	Java   = Language{"java"}
+	Python = Language{"python"}
+	NodeJS = Language{"nodejs"}
+	YAML   = Language{"yaml"}
+)
+
+// All providers a deterministic iteration of all valid languages.
+func All() iter.Seq[Language] { return all }
+
+// all is the implementation of [All].
+func all(yield func(Language) bool) {
```

Comment:
IMO a lazy iterator is a bit overkill here and makes the code in other parts a bit more complex. E.g. the hard-coding of the number of languages.

Feel free to disregard as it's just a nit, but I feel like just exposing the  langOrder array directly would be fine.